src/client.rs

```diff
@@ -59,8 +59,13 @@ pub struct rustls_client_session {
     _private: [u8; 0],
 }
 
+pub struct Sess {
```

Comment:
This type name is a little terse, I don't think it needs to be exported from the crate, and seems virtually identical to `server::Sess` so WDYT about having this in `session.rs`:
```
struct Session<S: rustls::Session> {
    session: S,
    userdata: *mut c_void,
}
```
[`rustls::ClientSession`](https://docs.rs/rustls/0.19.1/rustls/struct.ClientSession.html#impl-Session) and [`rustls::ServerSession`](https://docs.rs/rustls/0.19.1/rustls/struct.ServerSession.html#impl-Session) both implement [`rustls::Session`](https://docs.rs/rustls/0.19.1/rustls/trait.Session.html)

Reply:
Unfortunately then we wouldn't be able to call the methods on each that are specific to `Client` or `Server`, like `ServerSession::get_sni_hostname`.

Making this pub(crate) instead of pub is a good idea though.